File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPGravity.java
package org.noxet.noxetserver.playerstate.properties;

import org.bukkit.entity.Player;
import org.noxet.noxetserver.playerstate.PlayerStateProperty;

public class PSPGravity implements PlayerStateProperty<Boolean> {
    @Override
    public String getConfigName() {
        return "gravity";
    }

    @Override
    public Boolean getDefaultSerializedProperty() {
        return true;
    }

    @Override
    public Boolean getSerializedPropertyFromPlayer(Player player) {
        return player.hasGravity();
    }

    @Override
    public void restoreProperty(Player player, Boolean gravity) {
        player.setGravity(gravity);
    }
}

File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPGravity.scala
package org.noxet.noxetserver.playerstate.properties

import org.bukkit.entity.Player
import org.noxet.noxetserver.playerstate.PlayerStateProperty

object PSPGravity extends PlayerStateProperty[Boolean]:
  override def getConfigName: String = "gravity"

  override def getDefaultSerializedProperty: Boolean = true

  override def getSerializedPropertyFromPlayer(player: Player): Boolean = player.hasGravity

  override def restoreProperty(player: Player, gravity: Boolean): Unit =
    player.setGravity(gravity)